src/jloptions.c

```diff
@@ -505,6 +508,20 @@ JL_DLLEXPORT void jl_parse_opts(int *argcp, char ***argvp)
             else
                 jl_errorf("julia: invalid argument to --handle-signals (%s)", optarg);
             break;
+        case 's':
+            {
+                int s = strtol(optarg, &endptr, 10);
+                if (errno != 0 || optarg == endptr || *(endptr + 1) != 0) {
```

Comment:
Portability apart, it is wrong to use errno to check for error. It should only be used to check which error happens if the return value indicates that one does happen.